packages/http-framework/src/lib/Client.ts

```diff
@@ -52,29 +54,68 @@ export class Client extends EventEmitter {
 	 */
 	public async listen({ serverOptions, postPath, port, address, ...listenOptions }: ListenOptions) {
 		const key = await makeKey(this.#discordPublicKey);
+		const path = postPath ?? process.env.HTTP_POST_PATH ?? '/';
 
-		this.server = Fastify(serverOptions);
-		this.server.addHook('onRoute', (options) => assignRawData(options));
-		this.server.addHook('preHandler', (request, reply) => handleSecurityHook(request, reply, key));
-		this.server.post(postPath ?? process.env.HTTP_POST_PATH ?? '/', this.handleHttpMessage.bind(this));
+		this.server = createServer(serverOptions ?? {});
+		this.server.on('request', (request, response) => this.handleRawHttpMessage(request, response, path, key));
 
-		await this.server.listen({ ...listenOptions, port, host: address });
+		return new Promise<void>((resolve) => this.server.listen({ ...listenOptions, port, host: address }, resolve));
```

Comment:
Doesn't this silence errors that happen when listening?

Reply:
Errors are thrown synchronously, `listen(option, cb)` does not pass any parameters (not even an error) to the callback.